src/pykeen/training/training_loop.py

```diff
@@ -635,8 +635,11 @@ def _train(  # noqa: C901
                 self._epoch = epoch
 
                 should_stop = False
-                if stopper is not None and stopper.should_evaluate(epoch) and stopper.should_stop(epoch):
-                    should_stop = True
+                if stopper is not None and stopper.should_evaluate(epoch):
+                    if stopper.should_stop(epoch):
+                        should_stop = True
+                    # Since the model is also used within the stopper, its graph and cache have to be cleared
+                    self._free_graph_and_cache()
```

Comment:
It still seems odd to me that this is necessary, but if it fixes the issue :shrug: :smile: